api/v2/rpc.go

```diff
@@ -164,93 +193,23 @@ func (s *NotificationServer) DeleteNotification(ctx context.Context, req *pb.Del
 		return nil, status.Error(codes.InvalidArgument, "Failed to provide notification name")
 	}
 
-	err := s.Store.DeleteNotification(req.GetName())
+	tx, err := s.Store.Begin()
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
```

Comment:
Rollback() from database/sql actually is safe to do this and is considered an idiomatic pattern to do so. I think it makes sense both ways, though.